library/run_cmds.py

```diff
@@ -102,13 +101,14 @@ def __init__(self, module):
         if isinstance(self.fact_names, list):
             for f_name in self.fact_names:
                 f_name_list = f_name.strip().split('_', 1)
-                command = DEFAULT_CMD_DICT[f_name_list[0]]
-                fact = f_name_list[1]
-                if command in DEFAULT_CMDS:
-                    if command in self.facts_requested.keys():
-                        self.facts_requested[command].append(fact)
-                    else:
-                        self.facts_requested[command] = [fact]
+                if f_name_list[0] in DEFAULT_CMD_DICT:
+                    command = DEFAULT_CMD_DICT[f_name_list[0]]
+                    fact = f_name_list[1]
+                    if command in DEFAULT_CMDS:
```

Comment:
So likely not filed as a bug because probably 99% of use cases just run with the default fact list, you could pass in a "fact" that doesn't link to any of the fact commands in DEFAULT_CMD_DICT (i.e. Fake_fake.fact). This would have caused a problem where `command` isn't a real command to run. I encountered this when I pulled out `"Username": uname_command.UnameRhoCmd` from the dictionary but those facts are still being requested to be collected (they are just no longer gathered by RunCmd). Hopefully the explanation helps. Ultimately this file will go away once all commands are converted to `raw` tasks.